x/liquidity/spec/03_state_transitions.md

```diff
@@ -1,174 +1,235 @@
-  - Find `swapPrice(k)` which has the largest amount of X coins matched
+
+  - If `EX` > `EY`*`swapPrice`: Residual X order amount remains
+
+    - Amount of X coins matched from swap orders = `EY`*`swapPrice`
+    - Amount of Y coins matched from swap orders = `EY`
+
+  - If `EY` > `EX`/`swapPrice`: Residual Y order amount remains
+
+    - Amount of X coins matched from swap orders = `EX`
+    - Amount of Y coins matched from swap orders = `EX`/`swapPrice`
+
+### Increase case
+
+Iteration: iterate `orderPrice(i)` of all swap orders from low to high.
+
+Variables:
+
+- `EX(i)`: Sum of all order amount of swap orders that swap X for Y with order price equal or higher than this `orderPrice(i)`
+- `EY(i)`: Sum of all order amounts of swap orders that swap Y for X with order price equal or lower than this `orderPrice(i)`
+
+- ExactMatch: SwapPrice is found between two orderPrices
+
+  - `swapPrice(i)` = (`X` + 2_`EX(i)`)/(`Y` + 2_`EY(i-1)`)
+
+    - condition1) `orderPrice(i-1)` < `swapPrice(i)` < `orderPrice(i)`
+
+  - `PoolY(i)` = (`swapPrice(i)`_`Y` - `X`) / (2_`swapPrice(i)`)
+
+    - condition2) `PoolY(i)` >= 0
+
+  - If both above conditions are met, `swapPrice` is the swap price for this iteration
+
+    - Amount of X coins matched = `EX(i)`
+
+  - If one of these conditions doesn't hold, go to FractionalMatch
+
+- FractionalMatch: SwapPrice is found at an orderPrice
+
+  - `swapPrice(i)` = `orderPrice(i)`
+  - `PoolY(i)` = (`swapPrice(i)`_`Y` - `X`) / (2_`swapPrice(i)`)
+  - Amount of X coins matched:
+
+    - `EX(i)` ← min[ `EX(i)`, (`EY(i)`+`PoolY(i)`)*`swapPrice(i)` ]
+
+- Find optimized swapPrice:
+
+  - Find `swapPrice(k)` that has the largest amount of X coins matched
+
     - this is our optimized swap price
     - corresponding swap result variables
+
       - `swapPrice(k)`, `EX(k)`, `EY(k)`, `PoolY(k)`
 
-**Decrease case**
-
-- Iteration : iterate `orderPrice(i)` of all swap orders from high to low
-
-  - variables
-    - `EX(i)` : Sum of all order amount of swap orders which swap X for Y with order price equal or higher than this `orderPrice(i)`
-    - `EY(i)` : Sum of all order amount of swap orders which swap Y for X with order price equal or lower than this `orderPrice(i)`
-  - ExactMatch : SwapPrice is found between two orderPrices
-    - `swapPrice(i)` = (`X` + 2*`EX(i)`)/(`Y` + 2*`EY(i-1)`)
-      - condition1) `orderPrice(i)` < `swapPrice(i)` < `orderPrice(i-1)`
-    - `PoolX(i)` = (`X` - `swapPrice(i)`\*`Y`)/2
-      - condition2) `PoolX(i)` >= 0
-    - If both above conditions are met, `swapPrice` is the swap price for this iteration
-      - Amount of Y coins matched = `EY(i)`
-    - If one of above conditions doesn’t hold, go to FractionalMatch
-  - FractionalMatch : SwapPrice is found at an orderPrice
-    - `swapPrice(i)` = `orderPrice(i)`
-    - `PoolX(i)` = (`X` - `swapPrice(i)`\*`Y`)/2
-    - Amount of Y coins matched :
-      - `EY(i)` ← min[ `EY(i)`, (`EX(i)`+`PoolX(i)`)/`swapPrice(i)` ]
+### Decrease case
```

Comment:
let's describe what Decrease case means (generally, we will benefit from context)